Open Judge System/OJS.Services/Data/ParticipantScores/ParticipantScoresDataService.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace OJS.Services.Data.ParticipantScores
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using EntityFramework.Extensions;
+
+    using OJS.Data.Models;
+    using OJS.Data.Repositories.Contracts;
+    using OJS.Services.Data.Participants;
+
+    public class ParticipantScoresDataService : IParticipantScoresDataService
+    {
+        private readonly IEfGenericRepository<ParticipantScore> participantScores;
+        private readonly IParticipantsDataService participantsData;
+
+        public ParticipantScoresDataService(
+            IEfGenericRepository<ParticipantScore> participantScores,
+            IParticipantsDataService participantsData)
+        {
+            this.participantScores = participantScores;
+            this.participantsData = participantsData;
+        }
+
+        public ParticipantScore GetByParticipantIdAndProblemId(int participantId, int problemId) =>
+            this.participantScores
+                .All()
+                .FirstOrDefault(ps => ps.ParticipantId == participantId &&
+                    ps.ProblemId == problemId);
+
+        public ParticipantScore GetByParticipantIdProblemIdAndIsOfficial(int participantId, int problemId, bool isOfficial) =>
+            this.participantScores
+                .All()
+                .FirstOrDefault(ps => ps.ParticipantId == participantId &&
+                    ps.ProblemId == problemId &&
+                    ps.IsOfficial == isOfficial);
+
+        public IQueryable<ParticipantScore> GetAllQuery() =>
```

Comment:
IMO `Query` is redundant in the name, because there is no way to return the entity